3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/EnrichmentManager.scala

```diff
@@ -265,6 +265,31 @@ object EnrichmentManager {
       }
     } yield unitSuccess).getOrElse(unitSuccess)
 
+    // Calculate the derived timestamp
+    val derivedTstamp = EE.getDerivedTimestamp(
+      Option(event.dvce_sent_tstamp),
+      Option(event.dvce_created_tstamp),
+      Option(event.collector_tstamp),
+      Option(event.true_tstamp)
+    ) match {
+      case Success(dt) => {
+        dt.foreach(event.derived_tstamp = _)
+        unitSuccess
+      }
+      case f => f
+    }
+
+    // Fetch IAB enrichment context (before anonymizing the IP address)
+    val iabContext = registry.getIabEnrichment match {
```

Comment:
This can be rewritten as 
```scala
registry.getIabEnrichment.map { iab =>
  iab.getIabContext(???)
}
```

To avoid `.map(_.some)`.

This is a nit however. Feel free to ignore.